src/app/fromlove/[id]/page.tsx

```diff
@@ -0,0 +1,18 @@
+"use client";
+
+import { FromLoveDescription } from "@/components/FromLoveDescription/FromLoveDescription";
+import { PetHistoryItem } from "@/components/PetHistoryItem/PetHistoryItem";
+import { petsData } from "@/data/petsData";
+
+import { useParams } from "next/navigation";
+
+export default function FromLovePage() {
+  const { id } = useParams();
+  const pet = petsData.find((pet) => pet.name === id);
```

Comment:
Instead of using `useParams` on the client, you could leverage **Next.js dynamic route parameters** directly in the server-side page file. For example:
```js
interface FromLovePageProps {
    params: { id: string };
}

export default function FromLovePage({ params }: FromLovePageProps) {
    const { id } = params;
    const pet = petsData.find((pet) => pet.name === id);

    return (
        <div className="mt-20">
            {pet ? <PetHistoryItem name={pet.name} /> : <FromLoveDescription />}
        </div>
    );
}
```